HyphaApp/hypha
# Convert `STAFF_UPLOAD_CONTRACT` to Wagtail setting

To make it easier for administrators to change the Project workflow, change the low-level setting `STAFF_UPLOAD_CONTRACT` into a Wagtail Project Setting `staff_upload_contract`.

Issue #3952

## Test Steps

Before the change: only the Contracting role should be able to upload contracts to a project in Contracting status when system setting `STAFF_UPLOAD_CONTRACT` is `False` or not set.
After the change: the same should be true but only when the Wagtail Project Setting `staff_upload_contract` is `False` or not set, system setting `STAFF_UPLOAD_CONTRACT` should have no effect.

- Set up the sample database.
- As Staff, look at project 1 (`apply/projects/1/`), it should be in Contracting state and there should be no upload button available.
- As Contracting, look at project 1 (`apply/projects/1/`), it should be in Contracting state and there should be an upload button available.
- As Admin, go to Wagtail Admin -> Project Settings, check the Staff Upload Contract checkbox, and save.
- As Staff, look at project 1 again, it should be in Contracting state and there should be an upload button available.
- As Contracting, look at project again (`apply/projects/1/`), it should be in Contracting state and there should be an upload button available.
- As Staff, upload a PDF from `apply/projects/1/`, it should succeed.
